src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -97,33 +97,7 @@ public ValueTask ConnectAsync(EndPoint remoteEP, CancellationToken cancellationT
 
             saea.RemoteEndPoint = remoteEP;
 
-            ValueTask connectTask = saea.ConnectAsync(this, saeaCancelable: cancellationToken.CanBeCanceled);
-            if (connectTask.IsCompleted || !cancellationToken.CanBeCanceled)
-            {
-                // Avoid async invocation overhead
```

Comment:
was this lost somewhere in the transition? e.g. could there be perf impact? I would think this would be mostly UDP and maybe TCP to loopback where everything is fast...?